Add support in spatialite for librttopo

bigfootjon opened the pull request:
This is needed for Django's support of spatialite. Without `librttopo` it crashes while attempting to initialize a new spatialite DB, with this feature/library enabled it works.

I apologize as this probably isn't the correct way to do this, but I couldn't find any documentation that clarified this information. Some mistakes I probably made:

1. 2 packages changed in one PR (I split them up into 2 different commits)
2. Assigning a project that I am not a member of as the maintainer for the new `librttopo` project
3. `pkgcheck scan` reports several issues on `spatialite`  about "NonsolvableDepsInDev" and "NonsolvableDepsInStable" that I don't understand how to solve (maybe the library needs to be unmasked?)
4. General ignorance in how to write/update ebuilds
5. Unknown (to me) mistakes I've made, sorry in advance

AndrewAmmerlaan commented:
> 3\. `pkgcheck scan` reports several issues on `spatialite`  about "NonsolvableDepsInDev" and "NonsolvableDepsInStable" that I don't understand how to solve (maybe the library needs to be unmasked?)

This happens because we have `KEYWORDS="amd64 ~arm arm64 ~ia64 ~ppc ~ppc64 ~riscv x86"` in `spatialite.ebuild` but `KEYWORDS="~amd64 ~x86"` in `librttopo.ebuild`. To remove the error we need to add `dev-db/spatialite rttopo` to the `package.use.mask` files of the missing architectures (arm, arm64, powerpc, riscv, ia64).

Additionally since `spatialite.ebuild` has the stable `amd64` keyword, but `librttopo.ebuild` has the unstable `~amd64` keyword, we will need to `package.use.stable.mask` the new flag on `amd64` as well.

Did you also test this on `x86`? Otherwise please remove this keyword and leave only `amd64`.

AndrewAmmerlaan commented on sci-geosciences/librttopo/librttopo-1.1.0.ebuild:
```diff
@@ -0,0 +1,23 @@
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+src_configure() {
+	./autogen.sh
+	econf
+}
```
`autotools.eclass` provides the Gentoo specific `eautoreconf` function that probably should be used here instead of `./autogen.sh`. This should be run in the prepare phase (after applying (user) patches).

```suggestion
src_prepare() {
	default
	eautoreconf
}
```

bigfootjon commented:
@AndrewAmmerlaan Thank you very much for the review

> Did you also test this on x86? Otherwise please remove this keyword and leave only amd64.

Oops yeah I did not test on x86, removed!

All your other review comments I've fixed (I think)

AndrewAmmerlaan commented on dev-db/spatialite/spatialite-5.0.1-r2.ebuild:
```diff
@@ -1,4 +1,4 @@
+# Copyright 1999-2023 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
 EAPI=7
```
Since we have to revision bump this anyway, maybe we can bump the EAPI to 8 as well.

AndrewAmmerlaan commented on profiles/arch/amd64/package.use.stable.mask:
```diff
@@ -131,3 +131,7 @@ gnustep-base/gnustep-make libobjc2
 # Don't apply stable masks to python-exec since we're forcing every
 # impl there anyway. Please keep this in sync with use.stable.mask.
 dev-lang/python-exec -python_targets_pypy3
+
```
New entries to these files should go at the top.

AndrewAmmerlaan commented on sci-geosciences/librttopo/librttopo-1.1.0.ebuild:
```diff
@@ -0,0 +1,25 @@
+
+src_install() {
+	default
+	find "${D}" -name '*.la' -delete || die
```
```suggestion
	find "${ED}" -name '*.la' -delete || die
```

AndrewAmmerlaan commented on profiles/arch/arm64/package.use.mask:
```diff
@@ -435,3 +435,7 @@ sci-libs/linux-gpib firmware
+
+# Jon Janzen <jon@jonjanzen.com> (2023-09-11)
+# librttopo has not been tested on this arch
+dev-db/spatializte rttopo
```
I'm a bit confused why we still get the `NonsolvableDepsInStable` error on `arm64`, the flag is masked so it should work.

AndrewAmmerlaan replied:
> I'm a bit confused why we still get the `NonsolvableDepsInStable` error on `arm64`, the flag is masked so it should work.

aaah, it's because there is a type: `dev-db/spatializte` -> `dev-db/spatialite`.

AndrewAmmerlaan reviewed (commented):
Almost there, just some minor comments:
```
UnusedInherits | unused eclass: flag-o-matic
```
We can get rid of the `inherit flag-o-matic` since we are not using it.

```
find "${D}" -name '*.la' -delete || die
```
This should be:
```
find "${ED}" -name '*.la' -delete || die
```
To take the `EPREFIX` into account.

bigfootjon commented:
@AndrewAmmerlaan Thanks for reviewing, I've fixed everything you noticed and it looks like gentoo-repo-qa-bot is finally happy!

AndrewAmmerlaan commented:
Nice work, Thanks :+1: 

There was a dependency missing on `sci-libs/geos`, I took the liberty of amending this in before merging.
